app/src/main/java/org/simple/clinic/scanid/ScanSimpleIdUpdate.kt

```diff
@@ -55,12 +58,39 @@ class ScanSimpleIdUpdate @Inject constructor(
     if (model.isSearching) return noChange()
 
     return try {
-      val bpPassportCode = UUID.fromString(event.text)
-      val identifier = Identifier(bpPassportCode.toString(), BpPassport)
-      next(model = model.searching(), SearchPatientByIdentifier(identifier))
-    } catch (e: IllegalArgumentException) {
-      crashReporter.report(e)
-      noChange()
+      val patientPrefillInfoDetail = parseJsonIntoObject(event.text)
+      if (patientPrefillInfoDetail != null) {
+        val identifier = Identifier(patientPrefillInfoDetail.healthIdNumber, IndiaNationalHealthId)
+        next(model = model.searching(), SearchPatientByIdentifier(identifier))
+      } else {
+        noChange()
+      }
+    } catch (e: Exception) {
+      try {
+        val bpPassportCode = UUID.fromString(event.text)
+        val identifier = Identifier(bpPassportCode.toString(), BpPassport)
+        next(model = model.searching(), SearchPatientByIdentifier(identifier))
+      } catch (e: IllegalArgumentException) {
+        crashReporter.report(e)
+        noChange()
+      }
+    }
+  }
+
+  private fun parseJsonIntoObject(text: String): PatientPrefillInfo? {
+    val adapter = moshi.adapter(PatientPrefillInfo::class.java)
+    val patientPrefillInfoDetail: PatientPrefillInfo? = adapter.fromJson(text)
+    return patientPrefillInfoDetail?.let {
```

Comment:
We already have `PatientPrefillInfo?` above, so is there a reason we need to construct it again?